packages/flutter_tools/test/commands.shard/permeable/analyze_once_test.dart

```diff
@@ -241,9 +242,7 @@ flutter_project:lib/
         toolExit: true,
       );
     } finally {
-      if (optionsFile.existsSync()) {
```

Comment:
How about replacing https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/src/common.dart#L34 and using that not just for directories but also for files?

Reply:
That probably needs to be fixed too, but it is more for file lock issues on windows than read-only problems

Reply:
So, the difference would be a different error code? The same code structure could handle the situation apart from that, though, right? Maybe not for this PR, but it would be nice to have a single way for the tool to try to delete something.

Reply:
I think we would prefer if the tests shut down cleanly, and this was added as more of a stopgap

Reply:
Ah, yeah. Good point.